core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4QueryHttpSigner.java

```diff
@@ -31,11 +55,135 @@ public final class DefaultAwsV4QueryHttpSigner implements AwsV4QueryHttpSigner {
+    }
+
+    private SdkHttpRequest.Builder doSign(SignRequest<?, ? extends AwsCredentialsIdentity> request, String contentHash) {
+        SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();
+
+        Instant requestSigningInstant = validatedProperty(request, REQUEST_SIGNING_INSTANT);
+        Instant expirationInstant = validatedProperty(request, EXPIRATION_INSTANT,
+            requestSigningInstant.plusSeconds(SignerConstant.PRESIGN_URL_MAX_EXPIRATION_SECONDS));
+        Boolean doubleUrlEncode = validatedProperty(request, DOUBLE_URL_ENCODE, true);
+        Boolean normalizePath = validatedProperty(request, NORMALIZE_PATH, true);
+        String formattedRequestSigningDate = formatDateStamp(requestSigningInstant);
+        String formattedRequestSigningDateTime = formatTimestamp(requestSigningInstant);
+        String regionName = validatedProperty(request, REGION_NAME);
+        String serviceSigningName = validatedProperty(request, SERVICE_SIGNING_NAME);
+        String scope = buildScope(formattedRequestSigningDate, serviceSigningName, regionName);
+
+        long expirationInSeconds = getSignatureDurationInSeconds(requestSigningInstant, expirationInstant);
+
+        addHostHeader(requestBuilder);
+
+        AwsCredentialsIdentity sanitizedCredentials = sanitizeCredentials(request.identity());
+        if (sanitizedCredentials instanceof AwsSessionCredentialsIdentity) {
+            // For SigV4 query-signing, we need to add "X-Amz-Security-Token"
+            // as a query string parameter, before constructing the canonical
+            // request.
+            addSessionCredentials(requestBuilder, (AwsSessionCredentialsIdentity) sanitizedCredentials);
+        }
+
+        CanonicalRequest canonicalRequest =
+            createCanonicalRequest(request.request(), requestBuilder, contentHash, doubleUrlEncode,
+                normalizePath);
+
+        String canonicalRequestString = canonicalRequest.string();
+        String stringToSign = buildStringToSign(
+            canonicalRequestString,
+            AWS4_SIGNING_ALGORITHM,
+            formattedRequestSigningDateTime,
+            scope
+        );
+
+        byte[] signingKey = deriveSigningKey(
+            sanitizedCredentials,
+            requestSigningInstant,
+            regionName,
+            serviceSigningName
+        );
+
+        byte[] signature = computeSignature(stringToSign, signingKey);
+
+        String signingCredentials = sanitizedCredentials.accessKeyId() + "/" + scope;
+        requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_ALGORITHM, SignerConstant.AWS4_SIGNING_ALGORITHM);
+        requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_DATE, formattedRequestSigningDateTime);
+        requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_SIGNED_HEADERS, canonicalRequest.signedHeaderString());
+        requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_EXPIRES, Long.toString(expirationInSeconds));
```

Comment:
hmm, if these are being added later and not part of the canonicalRequestString, that feels wrong.

Reply:
yup, the test is failing in the CI.

Reply:
Fixed.